control-plane/pkg/reconciler/channel/controller.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package channel
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+	kafkachannelreconciler "knative.dev/eventing-kafka-broker/control-plane/pkg/client/injection/reconciler/messaging/v1beta1/kafkachannel"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	podinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/pod"
+	secretinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/secret"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/resolver"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/base"
+)
+
+func NewController(ctx context.Context, watcher configmap.Watcher, configs *Configs) *controller.Impl {
+
+	reconciler := &Reconciler{
+		Reconciler: &base.Reconciler{
+			KubeClient:                  kubeclient.Get(ctx),
+			PodLister:                   podinformer.Get(ctx).Lister(),
+			SecretLister:                secretinformer.Get(ctx).Lister(),
+			DataPlaneConfigMapNamespace: configs.DataPlaneConfigMapNamespace,
+			DataPlaneConfigMapName:      configs.DataPlaneConfigMapName,
+			DataPlaneConfigFormat:       configs.DataPlaneConfigFormat,
+			SystemNamespace:             configs.SystemNamespace,
+			DispatcherLabel:             base.ChannelDispatcherLabel,
+			ReceiverLabel:               base.ChannelReceiverLabel,
+		},
+		ClusterAdmin: sarama.NewClusterAdmin,
+		Configs:      configs,
+	}
+
+	impl := kafkachannelreconciler.NewImpl(ctx, reconciler)
+
+	reconciler.Resolver = resolver.NewURIResolver(ctx, impl.EnqueueKey)
```

Comment:
wanna use like `resolver.NewURIResolverFromTracker(ctx, impl.Tracker)` , see https://github.com/knative/eventing/commit/d879d093ac1f76c9d367f4217ca781445a4fbbd0  for reference